psydac/core/bsplines.py

```diff
@@ -108,22 +143,51 @@ def basis_funs( knots, degree, x, span ):
     by using 'left' and 'right' temporary arrays that are one element shorter.
 
     """
-    left   = np.empty( degree  , dtype=float )
-    right  = np.empty( degree  , dtype=float )
-    values = np.empty( degree+1, dtype=float )
+    left = np.empty(degree, dtype=float)
+    right = np.empty(degree, dtype=float)
+
+    out[0] = 1.0
+    for j in range(0, degree):
+        left[j] = x - knots[span - j]
+        right[j] = knots[span + 1 + j] - x
+        saved = 0.0
+        for r in range(0, j + 1):
+            temp = out[r] / (right[r] + left[j - r])
+            out[r] = saved + right[r] * temp
+            saved = left[j - r] * temp
+        out[j + 1] = saved
+
+
```

Comment:
Fixed in https://github.com/pyccel/psydac/commit/c9eac22256b58bb6b2138192ad3cc5766ae2d143

Reply:
About me using two empty lines instead of one at the end of function definitions, it is because my IDE yells at me if I don't. It is part of PEP8 : [as shown here](https://www.python.org/dev/peps/pep-0008/#blank-lines)

Reply:
I see. We have been using a comment separator instead, i.e. an 80-char long `#==========` which is much faster to spot than two empty lines. Can you not change your IDE warning preferences?

Reply:
It's not too bothersome anyway so I can just ignore it